### DISCUSSION TITLE
`SpeakerNet` Speaker Identification: how to identify unknown speaker?

### DISCUSSION THREAD

**Discussion by JINHXu** [NVIDIA-NeMo/NeMo#2411] [Q&A] [answered]
Hi!

I am using the `SpeakerNet` model for Speaker Identification, it appears that in the inference stage, the model always returns a label, however, I was hoping to also identify that "a stranger" is "a stranger", i.e. for a speaker that is not in the training data, I expect the model to also realize that the speaker is `unknown`. But in this case, the model always returns the most similar label instead of `unknown`. I have a few thoughts myself (mentioned [here](https://github.com/NVIDIA/NeMo/discussions/2383#discussioncomment-920460)), and I was wondering how is everyone else handling the situation? Ideas would be much appreciated!

Cheers!
Xu

**Comment by JINHXu** [comment, marked as answer]
To answer my own question, this can be done by adding an `Unk` class of various random speakers (no overlap with other speakers in the training data). I went with recordings from `voxceleb`, 3-4 audio samples from each speaker, the results look good on recognizing unknown speakers. In case this is helpful to anyone else.

**Comment by tkcoding** [comment]
@JINHXu sorry for the late comment here.
I have the thought of predicting the speech with **probability output (logits)** and from there understand the threshold instead of getting max value of the probability. For e.g. if a **similar voice typically the max probability will be > 0.7** if model produce max probability < 0.5 then we can put this into "stranger" class.
